cpp/src/arrow/compute/kernels/codegen_internal.h

```diff
@@ -807,11 +885,30 @@ ArrayKernelExec SignedInteger(detail::GetTypeId get_id) {
   }
 }
 
-// Generate a kernel given a templated functor for base binary types
+// Generate a kernel given a templated functor for base binary types. Generates
+// a single kernel for binary/string and large binary / large string. If your
+// kernel implementation needs access to the specific type at compile time,
+// please use BaseBinarySpecific.
 //
 // See "Numeric" above for description of the generator functor
 template <template <typename...> class Generator, typename Type0, typename... Args>
 ArrayKernelExec BaseBinary(detail::GetTypeId get_id) {
```

Comment:
```suggestion
ArrayKernelExec GenerateForBaseBinary(detail::GetTypeId get_id) {
```

Reply:
I'll move these "generator-dispatchers" to a `generate::` namespace for clarity. I thought that `codegen::` was clear enough but I guess not.